### DISCUSSION TITLE
Setup a FreeBSD and OpenBSD CI with SourceHut

### DISCUSSION THREAD

**Discussion by XVilka** [vlang/v#11625] [Ideas]
[SourceHut](https://sourcehut.org) offers the CI service that [supports both FreeBSD and OpenBSD](https://man.sr.ht/builds.sr.ht/compatibility.md#freebsd). And it support integration with GitHub commits and pull requests through [dispatch service](https://man.sr.ht/dispatch.sr.ht/).

**Comment by medvednikov** [comment]
That'd be great. Do you have experience with it?

**Comment by XVilka** [comment]
@medvednikov you need to open an account there and then add YAML configuration files.
This is what we use for radare2: https://github.com/radareorg/radare2/tree/master/.builds
Since V is very small and portable you probably wouldn't need anything extra too.

**Comment by XVilka** [comment]
Aaand, it fails again: https://builds.sr.ht/~xvilka/job/171495#task-install-654
Happens quite often with V on BSD - please setup SourceHut CI.

**Comment by medvednikov** [comment]
I can't set it up because it requires access to private repos, and setting it up from another account is not possible.

We'll have our own FreeBSD CI soon.

**Comment by M4SSD35TRUCT10N** [comment]
@medvednikov This should be done within at least the time frame of the beta release or release candidate. I'd put it in the beta release frame and 0.2 series.
